pkg/steps/source.go

```diff
@@ -293,6 +296,9 @@ func buildFromSource(jobSpec *api.JobSpec, fromTag, toTag api.PipelineImageStrea
 			},
 		},
 	}
+	if pullSecret != nil {
```

Comment:
There is no reason to pass the whole *coreapi.Secret, just only to check if its nil or not. Is there any other reason why we can't use a bool instead?

Reply:
Maybe pointer/address is fine here?
The original arg for the program is `--pull-secret-path` which is used to init the secret object.
I feel passing the secret (pointer) through makes more sense than the path: such as an invalid path.